paddle/fluid/framework/paddle2cinn/cinn_compiler.cc

```diff
@@ -61,8 +61,8 @@ using ::cinn::hlir::framework::BuildScope;
 using ::cinn::hlir::framework::GraphCompiler;
 
 CinnCompiler* CinnCompiler::GetInstance() {
-  static CinnCompiler instance;
-  return &instance;
+  static CinnCompiler* instance = new CinnCompiler();
```

Comment:
The reason why I do not destruct `CinnCompiler` because the process would hang when exiting.

<img width="1124" alt="image" src="https://user-images.githubusercontent.com/32832641/167631638-b2255193-020b-47cc-b1e5-837c6de2bec6.png">

Reply:
Not sure whether it is a good fix, the best would be trying to fix underlying destruct bug. I would ask @wzzju to take a look.

Reply:
As we discuss offline, we can defer this discussion on this workaround fix. The reason why this bug occurs should be checked carefully after this PR. c.c. @zhhsplendid .